File: 20140129/fizzbuzz.rb
class FizzBuzz
  def initialize(min, max)
      print "hoge"
    @min = min
    @max = max
  end

  def fizz? num
    return num % 3 == 0
  end

  def buzz? num
    return num % 5 == 0
  end

  def fizzbuzz num
    if fizz?(num) and buzz?(num)
      p "FizzBuzz"
    elsif fizz?(num)
      p "Fizz"
    elsif buzz?(num)
      p "Buzz"
    else
      p num
    end
  end

  def run
    for i in @min..@max
      fizzbuzz i
    end
  end
end

fb = FizzBuzz.new(1, 100)
fb.run

File: 20140621/fizzbuzz.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>
#include <errno.h>

static void
print_usage (void)
{
  printf ("[USAGE] ./a.out min max\n");
}

int
main (int argc, char *argv[])
{
  int i;
  int min,max;

  if (argc != 3)
    goto ERROR;
  
  errno = ELAST;
  min = strtol(argv[1], NULL, 10);
  if (min == 0 && (errno == EINVAL || errno == ERANGE))
    goto ERROR;

  errno = ELAST;
  max = strtol(argv[2], NULL, 10);
  if (min == 0 && (errno == EINVAL || errno == ERANGE))
    goto ERROR;
  
  for (i = min; i <= max; i++)
    {
      if (i % 15 == 0)
        printf ("FizzBuzz\n");
      else if (i % 3 == 0)
        printf ("Fizz\n");
      else if (i % 5 == 0)
        printf ("Buzz\n");
      else
        printf ("%d\n", i);
    }
  return 0;

 ERROR:
  print_usage ();
  exit (EXIT_FAILURE);
}
